### PR TITLE
Fix #1738: Re-introduce auto-play setting

### LINKED ISSUE #1738: Re-Enable the Auto-Play Toggle
Should be re-enabled by reverting https://github.com/brave/brave-ios/pull/1737 post 1.13 release for QA to properly test

Refs: https://github.com/brave/brave-ios/issues/1129, https://github.com/brave/brave-ios/pull/1735

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #1738 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

Go to https://edition.cnn.com/videos/world/2021/08/25/ramstein-air-base-afghanistan-evacuees-tent-city-shubert-pkg-intl-vpx.cnn

Verify that the video does not auto play with the setting on.

Keep in mind that not all videos on the internet work with this autoplay toggle, the url i posted seems to work reliably for testing.

## Screenshots:

Zrzut ekranu 2021-08-25 o 13 36 12

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/TabManager.swift [hunk 1]
```diff
@@ -164,9 +164,9 @@ class TabManager: NSObject {
         configuration.preferences.javaScriptCanOpenWindowsAutomatically = !Preferences.General.blockPopups.value
         configuration.dataDetectorTypes = .all
         
-//        if !Preferences.General.mediaAutoPlays.value {
-//            configuration.mediaTypesRequiringUserActionForPlayback = .all
-//        }
+        if !Preferences.General.mediaAutoPlays.value {
+            configuration.mediaTypesRequiringUserActionForPlayback = .all
```
Comment (iccub) [inline Client/Frontend/Browser/TabManager.swift]: default seems to be audio only, for our case we can translate to have the preference as enabled by default

```
(lldb) po configuration.mediaTypesRequiringUserActionForPlayback == .all
false

(lldb) po configuration.mediaTypesRequiringUserActionForPlayback == .audio
true
```

**Review by Brandon-T — APPROVED** [review]

File: Client/Frontend/Settings/SettingsViewController.swift [hunk 2]
```diff
@@ -251,10 +251,14 @@ class SettingsViewController: TableViewController {
...
         
-        general.rows.append(
-            .boolRow(title: Strings.enablePullToRefresh,
-                     option: Preferences.General.enablePullToRefresh,
-                     image: #imageLiteral(resourceName: "settings-pull-to-refresh").template)
+        general.rows.append(contentsOf: [
+                .boolRow(title: Strings.enablePullToRefresh,
+                         option: Preferences.General.enablePullToRefresh,
+                         image: #imageLiteral(resourceName: "settings-pull-to-refresh").template),
+                .boolRow(title: Strings.mediaAutoPlays,
```
Comment (Brandon-T) [inline Client/Frontend/Settings/SettingsViewController.swift:258]: I'm not 100% sure, but I think we need to add:
`Preferences.General.mediaAutoPlays.observe(from: self)` to `fileprivate func didInit()`
I missed this in my initial review and my own code for background playback preference. Not sure if needed here too.

**Review by Brandon-T — CHANGES_REQUESTED** [review]

File: Client/Frontend/Settings/SettingsViewController.swift (hunk 2 above)
Comment (iccub) [reply Client/Frontend/Settings/SettingsViewController.swift:258]: Seems to work fine without it, UI updates correctly as well as the preference value

**Review by kylehickinson — APPROVED** [review]
Thanks for picking this up :)

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**